File: src/main/java/io/renren/modules/oss/cloud/OSSFactory.java
package io.renren.modules.oss.cloud;

import io.renren.common.utils.ApplicationContextHolder;
import io.renren.common.utils.ConfigConstant;
import io.renren.common.utils.Constant;
import io.renren.modules.sys.service.SysConfigService;

/**
 * 文件上传Factory
 *
 * @author ChunLiang Hu
 * @email davichi2009@gmail.com
 * @date 2017-03-26 10:18
 */
public final class OSSFactory {
    private static SysConfigService sysConfigService;

    static {
        OSSFactory.sysConfigService = (SysConfigService) ApplicationContextHolder.getBean("sysConfigService");
    }

    public static AbstractCloudStorageService build() {
        //获取云存储配置信息
        CloudStorageConfig config = sysConfigService.getConfigObject(ConfigConstant.CLOUD_STORAGE_CONFIG_KEY, CloudStorageConfig.class);

        if (config.getType() == Constant.CloudService.QINIU.getValue()) {
            return new QiniuAbstractCloudStorageService(config);
        } else if (config.getType() == Constant.CloudService.ALIYUN.getValue()) {
            return new AliyunAbstractCloudStorageService(config);
        } else if (config.getType() == Constant.CloudService.QCLOUD.getValue()) {
            return new QcloudAbstractCloudStorageService(config);
        }

        return null;
    }

}

File: src/main/java/io/renren/modules/oss/cloud/OSSFactory.kt
package io.renren.modules.oss.cloud

import io.renren.common.utils.ApplicationContextHolder
import io.renren.common.utils.ConfigConstant
import io.renren.common.utils.Constant
import io.renren.modules.sys.service.SysConfigService

/**
 * 文件上传Factory
 *
 * @author ChunLiang Hu
 * @email davichi2009@gmail.com
 * @date 2017-03-26 10:18
 */
object OSSFactory {
    private var sysConfigService: SysConfigService? = null

    init {
        OSSFactory.sysConfigService = ApplicationContextHolder.getBean("sysConfigService") as SysConfigService
    }

    fun build(): AbstractCloudStorageService? {
        //获取云存储配置信息
        val config = sysConfigService!!.getConfigObject(ConfigConstant.CLOUD_STORAGE_CONFIG_KEY, CloudStorageConfig::class.java)

        return when {
            config.type == Constant.CloudService.QINIU.value -> QiniuAbstractCloudStorageService(config)
            config.type == Constant.CloudService.ALIYUN.value -> AliyunAbstractCloudStorageService(config)
            config.type == Constant.CloudService.QCLOUD.value -> QcloudAbstractCloudStorageService(config)
            else -> null
        }

    }

}
